clap_derive/src/derives/subcommand.rs

```diff
@@ -131,8 +131,34 @@ fn gen_augment(
 
             match &*kind {
                 Kind::ExternalSubcommand => {
-                    quote_spanned! { kind.span()=>
-                        let app = app.setting(clap::AppSettings::AllowExternalSubcommands);
+                    let ty = match variant.fields {
+                        Unnamed(ref fields) if fields.unnamed.len() == 1 => &fields.unnamed[0].ty,
+
+                        _ => abort!(
+                            variant,
+                            "The enum variant marked with `external_attribute` must be \
```

Comment:
`external_attribute`?

Reply:
This is a widespread problem.  To keep this PR focused, I created #2679

Reply:
I actually don't see it anywhere except in this place. The only other place I see it is one ui test for this error

Reply:
Huh, either I searched for the wrong thing or overly broadly